### ISSUE TITLE
"vite build" doesn't work when node_modules/.vite doesn't exist, and type generation on.

### ISSUE THREAD

**Issue by peter50216** [mathe42/vite-plugin-comlink#19]
For example, on CI with a clean checkout that use vite-plugin-comlink, `vite build` directly won't work since vite doesn't generate the cacheDir in `build` and there's no previous run of `vite dev` that generates the cacheDir. But currently the typeSaveFile is always saved in the cacheDir.

This cause `vite build` to fail with:
```
[comlink] ENOENT: no such file or directory, open '<path>/node_modules/.vite/comlink.json'
error during build:
Error: ENOENT: no such file or directory, open '<path>/node_modules/.vite/comlink.json'
    at Object.openSync (node:fs:585:3)
    at writeFileSync (node:fs:2153:35)
    at writeTypeDefs (<path>/node_modules/.pnpm/vite-plugin-comlink@2.0.2_comlink@4.3.1+vite@2.5.10/node_modules/vite-plugin-comlink/dist/index.js:77:33)
    at Object.buildEnd (<path>/node_modules/.pnpm/vite-plugin-comlink@2.0.2_comlink@4.3.1+vite@2.5.10/node_modules/vite-plugin-comlink/dist/index.js:82:7)
    at <path>/node_modules/.pnpm/rollup@2.56.2/node_modules/rollup/dist/shared/rollup.js:20199:25
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Promise.all (index 0)
    at async rollupInternal (<path>/node_modules/.pnpm/rollup@2.56.2/node_modules/rollup/dist/shared/rollup.js:20956:9)
    at async doBuild (<path>/node_modules/.pnpm/vite@2.5.10/node_modules/vite/dist/node/chunks/dep-36bf480c.js:51091:24)
    at async build (<path>/node_modules/.pnpm/vite@2.5.10/node_modules/vite/dist/node/chunks/dep-36bf480c.js:50932:16)
 ELIFECYCLE  Command failed with exit code 1.
```

Possible solution:
* Don't generate .d.ts on build (I'm not sure if it's expected to have anything outside of `dist` changed when building)
* Create the cache directory by ourself if it doesn't exist.
* Don't write the typeSaveFile if the cacheDir doesn't exist.

(I'd prefer the first option.)

**Comment by mathe42** [comment]
I opt for 3. as while depeveloping I use build to get a full list of modules (they are only added when the specific module is requested) and not writing the cache would remove that benefit. for CI you can change the vite config to opt out.

**Comment by peter50216** [comment]
I'm fine with either solutions as long as it doesn't error out on clean build. Agree that it's convenient to have build regenerates all the type definitions.

Thanks for the timely responses 👍

**Comment by mathe42** [comment]
Released with version 2.1

**Comment by mathe42** [comment]
> I'm fine with either solutions as long as it doesn't error out on clean build. Agree that it's convenient to have build regenerates all the type definitions.
> 
> Thanks for the timely responses 👍

And thanks for your contributions!
